extensions/schema-compare/src/dialogs/schemaCompareOptionsDialog.ts

```diff
@@ -436,60 +440,76 @@ export class SchemaCompareOptionsDialog {
 	protected async execute() {
 		this.SetDeploymentOptions();
 		this.SetObjectTypeOptions();
+		this.disposeListeners();
 	}
 
 	protected async cancel() {
+		this.disposeListeners();
 	}
 
 	private async reset() {
 		let service = await azdata.dataprotocol.getProvider<azdata.SchemaCompareServicesProvider>('MSSQL', azdata.DataProviderType.SchemaCompareServicesProvider);
 		let result = await service.schemaCompareGetDefaultOptions();
 		this.deploymentOptions = result.defaultDeploymentOptions;
 
-		this.generaloptionsCheckBoxes.forEach(option => {
-			option.checked = this.GetSchemaCompareOptionUtil(option.label);
-		});
-		this.objectTypesCheckBoxes.forEach(obj => {
-			obj.checked = this.GetSchemaCompareIncludedObjectsUtil(obj.label);
-		});
+		this.updateOptionsTable();
+		this.optionsFlexBuilder.removeItem(this.optionsTable);
+		this.optionsFlexBuilder.insertItem(this.optionsTable, 0, { CSSStyles: { 'overflow': 'scroll', 'height': '68vh' } });
+
+		this.updateObjectsTable();
+		this.objectTypesFlexBuilder.removeItem(this.objectsTable);
+		this.objectTypesFlexBuilder.addItem(this.objectsTable, { CSSStyles: { 'overflow': 'scroll', 'height': '80vh' } });
 	}
 
 	private async initializeSchemaCompareOptionsDialogTab() {
 		this.generalOptionsTab.registerContent(async view => {
 
-			this.optionsFlexBuilder = view.modelBuilder.flexContainer()
-				.withLayout({
-					flexFlow: 'column',
-				}).component();
-
 			this.descriptionHeading = view.modelBuilder.table().withProperties({
 				columns: [
 					{
 						value: 'Option Description',
 						headerCssClass: 'no-borders',
 						toolTip: 'Option Description'
-					},
+					}
 				]
 			}).component();
 
 			this.descriptionText = view.modelBuilder.text().withProperties({
 				value: ' '
 			}).component();
 
-			this.GetGeneralOptionCheckBoxes(view);
-			this.generaloptionsCheckBoxes.forEach(box => {
-				this.optionsFlexBuilder.addItem(box);
-			});
 
-			let uberOptionsFlexBuilder = view.modelBuilder.flexContainer()
+			this.optionsTable = view.modelBuilder.table().withProperties({
+				data: []
+			}).component();
+
+			this.updateOptionsTable();
+
+			this.disposableListeners.push(this.optionsTable.onRowSelected(async () => {
+				let row = this.optionsTable.selectedRows[0];
+				let label = this.optionsLabels[row];
+				this.descriptionText.updateProperties({
+					value: this.GetDescription(label)
+				});
+			}));
+
+			this.disposableListeners.push(this.optionsTable.onCellAction(async (rowState) => {
+				let checkboxState = <azdata.ICheckboxCellActionEventArgs>rowState;
```

Comment:
IIUC, you are trying to check whether the eventargs is of type ICheckboxCellActionEventArgs, I am afraid this is not going to work, interface only exists at compile time, at run time you can only check instanceof class or check the desired properties explicitly

Reply:
Good point, I will add check for properties.  

I will keep the convert though - since .checked is not present in ICellActionArg and the compiler throws error if I don't convert. (I can do any but this seems more readable).